src/FileAdder/FileAdder.php

```diff
@@ -290,11 +277,14 @@ public function toMediaCollection(string $collectionName = 'default', string $di
         $media->file_name = $this->fileName;
 
         $media->disk = $this->determineDiskName($diskName, $collectionName);
-
         if (is_null(config("filesystems.disks.{$media->disk}"))) {
             throw DiskDoesNotExist::create($media->disk);
         }
 
+        $media->conversions_disk = $this->determineConversionsDiskName($media->disk, $collectionName);
+        if (is_null(config("filesystems.disks.{$media->conversions_disk}"))) {
```

Comment:
I've streamlined it a bit by moving the duplicated check to it's own function. I haven't put the check in `determineConversionsDiskName`, because then I'd lose the ability to early return.

If you feel this can be improved, feel free to commit it to this branch.

Reply:
> because then I'd lose the ability to early return.

My thinking was that you'd throw from within `determineConversionsDiskName`, which would in essence be an early return without the clutter. But no problem to keep it like it is now!